rules/nat.go

```diff
@@ -18,64 +18,61 @@ import (
 	"fmt"
 	"sort"
 
+	log "github.com/sirupsen/logrus"
+
 	"github.com/projectcalico/felix/iptables"
 )
 
+func makeRule(r *DefaultRuleRenderer, protocol string, action iptables.Action, ipVersion uint8, rule iptables.Rule) iptables.Rule {
```

Comment:
Few points:

- Instead of passing  in the rule renderer as the first argument, you should make this a method:
```
def (r *DefaultRuleRenderer) makeRule(protocol string, action iptables.Action, ipVersion uint8, rule iptables.Rule) iptables.Rule
```

- The rule argument is superfluous so it can be removed.
- Since there are a lot of functions in the package, worth picking a more detailed name: `natOutgoingRule`, for example.  (I suggested `makeRule` because I was thinking of defining the function inside the `NATOutgoingChain` method, where a short name would be OK.  Nothing wrong with making a dedicated method instead but the name needs to be more descriptive.)

Reply:
ack